doc/developers-guide/Hooks-and-handlers.md

```diff
@@ -54,13 +53,10 @@ ListOfSomething = ejabberd_hooks:run_fold(a_certain_hook,
                                            StateData#state.server])
 ```
 
-In between the XMPP domain (`StateData#state.server`) and handler arguments the initial value of the accumulator being passed through the sequence of handlers is inserted - in this case an empty list (`[]`).
-All handlers attached to the hook have to accept a list as a first argument and also return a list - possibly the same list
+In between the XMPP domain (`StateData#state.server`) and handler arguments is the initial value of the accumulator being passed through the sequence of handlers is inserted - in this case an empty list (`[]`).
```

Comment:
The initial value of the accumulator being passed through the sequence of handlers (in this case an empty list (`[]`)) is inserted between the XMPP domain (`StateData#state.server`) and handler arguments.